src/main/java/programminglife/model/Graph.java

```diff
@@ -73,4 +73,13 @@ public String getId() {
     public int size() {
         return this.getNodes().size();
     }
+
+    /**
+     *
+     * @param node {@link Node} to check if it is contained in the graph.
+     * @return boolean with the result of the contains.
+     */
+    public boolean contains(Node node) {
+        return this.getNodes().contains(node);
```

Comment:
It would be faster to run this on the map (`this.nodes.contains()`) since that will use the hash function instead of first generating a Collection and then searching that.